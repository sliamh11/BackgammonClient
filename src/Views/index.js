import LoginView from './Login/LoginView';
import GameView from './Game/GameView';
import ContactView from './Contact/ContactView';
import ChatView from './Chat/ChatView';
import SignupView from './Signup/SignupView';

export{
    LoginView,
    GameView,
    ContactView,
    ChatView,
    SignupView
}