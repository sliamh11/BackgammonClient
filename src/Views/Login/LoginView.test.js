import { LoginView } from "Views";
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Do I really need this?
configure({ adapter: new Adapter() });

describe("<Login /> Elements test", () => {
    // Shallow - renders the component, without it's leaf nodes.
    const container = shallow(<LoginView />);

    it("should match the snapshot", () => {
        expect(container.html()).toMatchSnapshot();
    });

    it("Should have a username field", () => {
        expect(container.find(<input type="text" name="username" />));
    });

    it("Should have a password field", () => {
        expect(container.find(<input type="text" name="password" />));
    });

    it("Should have a button", () => {
        expect(container.find(<div className="Button" />));
    });
});

// Cant test functionallity because mount isn't valid for React 17.
// Mount renders the leaf nodes (input, button, etc) and allows me to send data to it,
// therefor allows me checking the response for different inputs.

// describe("<Login /> functional test", () => {
//     // mount - renders the Component's leaves too.
//     const container = mount(<LoginView></LoginView>)
//     console.log(container);
//     // it("should raise an error message", () => {
//     //     container.find(<input name="username"/>).simulate('change', {
//     //         taget: {
//     //             value: "ron"
//     //         }
//     //     });

//     //     expect(container.find(<div className="error"/>));
//     // })
// });
