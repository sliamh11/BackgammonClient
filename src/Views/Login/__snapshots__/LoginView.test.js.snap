// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Elements test should match the snapshot 1`] = `"<div class=\\"container center\\"><div class=\\"Box\\"><div class=\\"Form\\"><h1>Login</h1><h3></h3><div class=\\"content\\"><div class=\\"Input\\"><div><label class=\\"label\\">Username:</label></div><div class=\\"Line \\"><input type=\\"text\\" name=\\"username\\" value=\\"\\"/></div></div><div class=\\"Input\\"><div><label class=\\"label\\">Password:</label></div><div class=\\"Line \\"><input type=\\"text\\" name=\\"password\\" value=\\"\\"/></div></div><div class=\\"center\\"><div class=\\"Button\\"><div class=\\"Line \\" justify=\\"between\\"><div>Login</div></div></div></div></div></div></div></div>"`;
